### PR TITLE
Fix support for Floatwheel LCM and allow brightness tuning via FloatControl/Floaty

### PR DESCRIPTION
**Pull request by surfdado**

### CODE REVIEW

File: refloat/refloat/main.c [hunk 1]
```diff
@@ -2392,6 +2392,7 @@ static void lights_control_request(CfgLeds *leds, uint8_t *buffer, size_t len, L
         return;
     }
 
+    data *d = (data *) ARG;
```
Comment (lukash) [inline refloat/refloat/main.c]: Please don't do this (unless in special cases), just pass the needed struct pointer as an additional argument. See next comment though.

File: refloat/refloat/main.c [hunk 2]
```diff
@@ -2404,6 +2405,26 @@ static void lights_control_request(CfgLeds *leds, uint8_t *buffer, size_t len, L
...
+            uint8_t brightness_status = buffer[4];
+
+            if ((d->state.mode == MODE_NORMAL) && (d->state.state == STATE_RUNNING)) {
+                leds->headlights.brightness = (float)(brightness) / 100.0;
+                leds->taillights.brightness = (float)(brightness) / 100.0;
+            }
+            else {
+                leds->front.brightness = (float)(brightness_idle) / 100.0;
+                leds->rear.brightness = (float)(brightness_idle) / 100.0;
+            }
```
Comment (lukash) [inline refloat/refloat/main.c]: I don't see the point of this condition? You only allow changing headlights when running and only allow idle brightness when not running? You're setting independent values, why not just set both unconditionally?

File: refloat/refloat/main.c [hunk 3]
```diff
@@ -2404,6 +2405,26 @@ static void lights_control_request(CfgLeds *leds, uint8_t *buffer, size_t len, L
...
+            leds->status_idle.brightness = (float)(brightness_status) / 100.0;
```
Comment (lukash) [inline refloat/refloat/main.c:2416]: This is sadly not what you wanted, it only sets the brightness for the idle animation, which is just an optional cool thing, it's not the main brightness of the status indicators like battery / sensors / duty.

The "normal" status brightness is determined by two variables: `leds->status.brightness_headlights_on` and `leds->status.brightness_headlights_off`. One is used when headlights are turned on (meant to be set to low brightness for night riding) and the other for when they're off (meant to be set bright for daylight). I'm not actually sure how to resolve this...

File: refloat/refloat/main.c [hunk 4]
```diff
@@ -2404,6 +2405,26 @@ static void lights_control_request(CfgLeds *leds, uint8_t *buffer, size_t len, L
...
+        }
+
+        if (len > 7) {
+            // TODO: allow changing of headlight pattern, idle pattern, and status pattern/mode
+        }
```
Comment (lukash) [inline refloat/refloat/main.c:2421]: I'd prefer not to add these "future" marks, they aren't doing anything and therefore are just noise. Do you _need_ to reserve the rest of the message for something undefined? (IMO, as we already discussed a bit, the interface is going to be ultimately very different)

**Review by lukash — CHANGES_REQUESTED** [review]
The CI has failed on `clang-format`. I still need to add a `CONTRIBUTING` doc describing how to run this stuff. In short, there's a `lefthook` (https://github.com/evilmartians/lefthook) config for a pre-commit hook, you can use that and run `lefthook run clang-format-fix` in the repo to fix any formatting issues automatically (and do `lefthook install` to have the check run on every commit). Let me know how that works for you, I know `lefthook` is not that universally available in distros, etc.

As a quick solution you can also run `clang-format -i refloat/refloat/main.c`.

You need clang-format version 17 installed (package may be called `clang-tools` or something along those lines).

Since the first commit is maybe a bit problematic with the status brightness, would you mind sending a PR just with the brightness fix and put the other one to a separate PR?

**Comment by lukash** [comment]
Oh, one more thing. I'm trying to adopt a system for automatic changelog generation, it's a bit hairy at the moment, to my dismay. The way it works is you add a "Fix: ..." or a "Feature: ..." at the end of the commit message. It's called a git trailer, same as "Signed-off-by: " that you use. I then have a script (it's in the root of the package) that collects these and dumps them into a piece of markdown. It unfortunately doesn't handle multiline text well :disappointed: 

Anyway, could you add a "Fix: Fix setting brightness for LCM modules." or something along those lines to the end of the commit message of the fix? Thanks!

File: refloat/refloat/main.c (hunk 2 above)
Comment (surfdado) [reply refloat/refloat/main.c]: yes, good point

**Pull request closed without merging**